### PR TITLE
Adds ccache to build workflow

### PR DESCRIPTION
**Pull request by cppcooper**
I tested on my fork with PR https://github.com/cppcooper/dfhack/pull/3

I added this commit and pushed, opened the PR, added a (temporary) document change commit and pushed. The second commit's build steps were seconds instead of minutes. So this works as far as I tested.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -60,6 +61,11 @@ jobs:
       with:
         path: ~/DF
         key: ${{ steps.env_setup.outputs.df_version }}
+    - name: Fetch ccache
+      uses: actions/cache@v2
+      with:
+        path: ~/.ccache
+        key: ${{ matrix.gcc }}
```
Comment (lethosor) [inline .github/workflows/build.yml:68]: Could this also include `${{ matrix.os }}`? That way we would use a clean cache if we move the GCC 7 build to a newer Ubuntu version, for instance.

More generally, I would tend towards more verbose keys to ensure that they are unique across all other caches in this pipeline, but that is probably not important for now. (One example: `ccache-${{ matrix.os }}-gcc-${{ matrix.gcc }}`)

**Comment by lethosor** [comment]
From Discord: looks like 20-30sec build times (with an existing cache) and a cache size of 12-15MB. Impressive results!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -38,7 +38,8 @@ jobs:
             libxml-libxslt-perl \
             lua5.3 \
             ninja-build \
-            zlib1g-dev
+            zlib1g-dev \
```
[the remark this answers is not in the document]
Comment (cppcooper) [reply .github/workflows/build.yml]: moved

File: .github/workflows/build.yml (hunk 1 above)
Comment (cppcooper) [reply .github/workflows/build.yml]: copy pasta'd

**Comment by myk002** [comment]
The cache will grow over time as old stuff becomes useless but is still cargo culted. Can we limit the size of the cache to working set size times the number of branches that it is building simultaneously? Let's say maybe 15mb * 50 = 750 mb ?

Edit: actually, ccache has a default limit of 1gb, which is close enough. No changes needed, I think

**Comment by myk002** [comment]
Alright, let's ship it and see how it goes!

**Pull request merged**